qemu/qmp.go

```diff
@@ -178,6 +178,28 @@ type MemoryDevices struct {
 	Type string            `json:"type"`
 }
 
+// CPUInfo represents information about each virtual CPU
+type CPUInfo struct {
+	CPU      int           `json:"CPU"`
+	Current  bool          `json:"current"`
+	Halted   bool          `json:"halted"`
+	QomPath  string        `json:"qom_path"`
+	Arch     string        `json:"arch"`
+	Pc       int           `json:"pc"`
+	ThreadID int           `json:"thread_id"`
+	Props    CPUProperties `json:"props"`
+}
+
+// CPUInfoFast represents information about each virtual CPU
```

Comment:
How is it different from `CPUInfo`? I'm asking since you provided the same comment here.

Reply:
They are basically same, one is returned by `query-cpus`, the other is returned by `query-cpus-fast`. And it has difference json format with `CPUInfo`.

Reply:
Reference: https://qemu.weilnetz.de/doc/qemu-qmp-ref.html

```
Command: query-cpus-fast
Returns information about all virtual CPUs. This command does not incur a performance penalty and should be used in production instead of query-cpus.

Returns: list of CpuInfoFast

Since: 2.12

Example:

-> { "execute": "query-cpus-fast" }
<- { "return": [
        {
            "thread-id": 25627,
            "props": {
                "core-id": 0,
                "thread-id": 0,
                "socket-id": 0
            },
            "qom-path": "/machine/unattached/device[0]",
            "arch":"x86",
            "target":"x86_64",
            "cpu-index": 0
        },
        {
            "thread-id": 25628,
            "props": {
                "core-id": 0,
                "thread-id": 0,
                "socket-id": 1
            },
            "qom-path": "/machine/unattached/device[2]",
            "arch":"x86",
            "target":"x86_64",
            "cpu-index": 1
        }
    ]
}
```